data/debug-cli/docker_debug.yaml

```diff
@@ -0,0 +1,166 @@
+      experimentalcli: false
+      kubernetes: false
+      swarm: false
+deprecated: false
+experimental: false
+experimentalcli: false
+kubernetes: false
+swarm: false
+examples: |-
+  ### Example 1: Get a shell into any image
+  This example pulls an nginx image and gets a shell into it without ever running a container.
+  
+  ```console
+  $ docker debug nginx
+  
+  Builtin commands:
+  - install [tool1] [tool2] ...    Add Nix packages from: https://search.nixos.org/packages
+  - uninstall [tool1] [tool2] ...  Uninstall NixOS package(s).
+  - entrypoint                     Print/lint/run the entrypoint.
+  - builtins                       Show builtin commands.
+  
+  Checks:
+  ✓ distro:            Debian GNU/Linux 12 (bookworm)
+  ✓ entrypoint linter: no errors (run 'entrypoint' for details)
+
+  Note: This is a sandbox shell. All changes will not affect the actual image.
+                                                                                                                     Version: 0.0.18 (BETA)
+  root@a8758716bb6a / [nginx:latest]
+  docker >
+  ```
+  
+  The console now has a shell in the read-only filesystem of the nginx image.
+  You can now run `entrypoint` to inspect the entrypoint or run `builtins` to
+  see all builtin commands.
+  
+  ### Example 2: Dumping a file from an image to stdout by using the -c mode
+  
+  Use the `-c` (`--command`) flag to directly evaluate a command instead of
+  running the shell in interactive mode.
+  
+  The following example dumps the default `index.html` file of nginx to stdout.
+
+  ```console
+  $ docker debug nginx -c "cat /usr/share/nginx/html/index.html"
+  <!DOCTYPE html>
+  <html>
+  <head>
+  <title>Welcome to nginx!</title>
+  <style>
+  html { color-scheme: light dark; }
+  body { width: 35em; margin: 0 auto;
+  font-family: Tahoma, Verdana, Arial, sans-serif; }
+  </style>
+  </head>
+  <body>
+  <h1>Welcome to nginx!</h1>
+  <p>If you see this page, the nginx web server is successfully installed and
+  working. Further configuration is required.</p>
+  
+  <p>For online documentation and support please refer to
+  <a href="http://nginx.org/">nginx.org</a>.<br/>
+  Commercial support is available at
+  <a href="http://nginx.com/">nginx.com</a>.</p>
+  
+  <p><em>Thank you for using nginx.</em></p>
+  </body>
+  </html>
+  ```
+  
+  ### Example 3: Get a shell into a stopped container via SSH
+  
+  Use the `--host` flag to connect to a different Docker Engine other than the
+  default.
+  
+  The following example connects to a remote Docker instance via SSH.
+  
+  ```console
+  $ docker debug --host ssh://root@example.org my-container
+  ```
+  
+  The `--host` flag may be used with all other examples, including getting a
+  shell into an image.
+  
+  ### Example 4: Get a shell into a running container that has no shell
+  
+  ```console
+  $ docker run --detach --rm --volume=/var/run/docker.sock:/var/run/docker.sock -p 8080:8080 --name my-container amir20/dozzle
```

Comment:
docs people, are there any other more popular images we can use here that doesn't include any shell, ie is a slim image? @mikesir87 any chance you know of some better image we could use here?

The problem with the image in the example is the `--volume=..` stuff. It's distracting and some people asked me if that is important for the example. But it's totally not. The intent of the example is to showcase that for slim images/containers `docker debug my-container` works while `docker exec -it my-container sh` does not.

Reply:
`hello-world` is very minimal, and an official image 😉

Reply:
Created an example with hello-world. The new example has the user use debug and then `cat /etc/shells` to show that there are no shells in the image. I'm not sure if they'll understand that. Is there a better way to show this?